core/src/wallet/ripple/database/RippleLikeTransactionDatabaseHelper.cpp

```diff
@@ -0,0 +1,137 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+
+#include "RippleLikeTransactionDatabaseHelper.h"
+#include <database/soci-option.h>
+#include <database/soci-date.h>
+#include <database/soci-number.h>
+#include <crypto/SHA256.hpp>
+#include <wallet/common/database/BlockDatabaseHelper.h>
+
+using namespace soci;
+
+namespace ledger {
+    namespace core {
+
+        bool RippleLikeTransactionDatabaseHelper::getTransactionByHash(soci::session &sql,
+                                                                       const std::string &hash,
+                                                                       RippleLikeBlockchainExplorerTransaction &tx) {
+
+            rowset<row> rows = (sql.prepare << "SELECT  tx.hash, tx.value, tx.time, "
+                    " tx.sender, tx.receiver, tx.fees, tx.confirmations, "
+                    "block.height, block.hash, block.time, block.currency_name "
+                    "FROM ripple_transactions AS tx "
+                    "LEFT JOIN blocks AS block ON tx.block_uid = block.uid "
+                    "WHERE tx.hash = :hash", use(hash));
+
+            for (auto &row : rows) {
+                inflateTransaction(sql, row, tx);
+                return true;
+            }
+
+            return false;
+        }
+
+        bool RippleLikeTransactionDatabaseHelper::inflateTransaction(soci::session &sql,
+                                                                     const soci::row &row,
+                                                                     RippleLikeBlockchainExplorerTransaction &tx) {
+            tx.hash = row.get<std::string>(0);
+            tx.value = BigInt::fromHex(row.get<std::string>(1));
+            tx.receivedAt = row.get<std::chrono::system_clock::time_point>(2);
+            tx.sender = row.get<std::string>(3);
+            tx.receiver = row.get<std::string>(4);
+            tx.fees = BigInt::fromHex(row.get<std::string>(5));
+            tx.confirmations = get_number<uint64_t>(row, 6);
+            if (row.get_indicator(7) != i_null) {
+                RippleLikeBlockchainExplorer::Block block;
+                block.height = get_number<uint64_t>(row, 7);
+                block.hash = row.get<std::string>(8);
+                block.time = row.get<std::chrono::system_clock::time_point>(9);
+                block.currencyName = row.get<std::string>(10);
+                tx.block = block;
+            }
+
+            return true;
+        }
+
+        bool RippleLikeTransactionDatabaseHelper::transactionExists(soci::session &sql,
+                                                                    const std::string &rippleTxUid) {
+            int32_t count = 0;
+            sql << "SELECT COUNT(*) FROM ripple_transactions WHERE transaction_uid = :rippleTxUid", use(rippleTxUid), into(
```

Comment:
While this is technically correct, the performance of this call will depend on the RDBMS’ optimizer (and in the case of SQLite… 🙄). What you could do is simply `SELECT 1` and check if the return result-set is empty or not.

Reply:
Hmm not really sure about that, first `SELECT 1` will expand column list and then pull all data, second if we are talking about `COUNT(*)` and `COUNT(1)` there is no difference in term of performance, this assumption might be true in old RDBMS